### PR TITLE
[widgets] use `chatCompletionStream`

### PR DESCRIPTION
**Pull request by mishig25**
Thanks to https://github.com/huggingface/huggingface.js/pull/645, it is now possible to use `chatCompletionStream` in Conversational Widget

image

### CODE REVIEW

**Comment by mishig25** [comment]
ready for review !

**Comment by radames** [comment]
nice @mishig25 LGTM! one curious question, how are you differentiating the models supporting chatCompletion API versus simple text generation?  this https://huggingface.co/bigscience/bloom-560m from https://huggingface.co/HuggingFaceH4/zephyr-7b-beta? Because the `chatCompletion` would throw an error in case the backend doesn't support the API,

File: packages/widgets/src/lib/components/InferenceWidget/widgets/ConversationalWidget/ConversationalWidget.svelte [hunk 1]
```diff
@@ -125,33 +117,18 @@
...
-			error = `An error occurred while rendering the chat template: "${(e as Error).message}"`;
-			return;
-		}
+		const previousMessages = [...messages];
 
-		const input: TextGenerationInput & Required<Pick<TextGenerationInput, "parameters">> = {
-			inputs: chatText,
-			parameters: {
-				return_full_text: false,
-			},
```
Comment (julien-c) [inline packages/widgets/src/lib/components/InferenceWidget/widgets/ConversationalWidget/ConversationalWidget.svelte:154]: given that's recent code from @Wauplin and @SBrandeis, i would like a review from them on this

**Comment by julien-c** [comment]
@radames i think the conversational widget would not be displayed on pure text-generation models like https://huggingface.co/bigscience/bloom-560m

**Comment by radames** [comment]
> @radames i think the conversational widget would not be displayed on pure text-generation models like https://huggingface.co/bigscience/bloom-560m

yes that's right, I'm just curious how on differentiate these models since they're all `pipeline_tag: text-generation` ?

**Comment by julien-c** [comment]
@radames there's a notion of a `WidgetType` that can be conversational:

https://github.com/huggingface/huggingface.js/blob/d1c0cb8604db4f504fa33b233fc12b9a501abc17/packages/tasks/src/pipelines.ts#L666

If a model repo is `text-generation` AND it has a tag of `conversational`, it displays that widget

cc @osanseviero who implemed

**Review by SBrandeis — APPROVED** [review]
Nice factorization!

**Pull request merged**